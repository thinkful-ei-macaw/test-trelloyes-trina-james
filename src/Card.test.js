import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card.js';

describe('My component', () => {
  it('renders without crashing', () => {
    // first create a DOM element to render the component into
    const div = document.createElement('div');
  
    // render the component, this is the actual test, if something is wrong it will fail here
    ReactDOM.render(<Card key='4' title='title' contnet='content'/>, div);
  
    // clean up code
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Card key='4' title='title' content='content'/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });

    it('renders the UI as expected', () => {
      const tree = renderer
        .create(<Card key='2' title='Bigger Title' content='Better Content'/>)
        .toJSON();
      expect(tree).toMatchSnapshot();  
      });
})