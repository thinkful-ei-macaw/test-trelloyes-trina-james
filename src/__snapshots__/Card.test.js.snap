// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My component renders the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    title
  </h3>
  <p>
    content
  </p>
</div>
`;

exports[`My component renders the UI as expected 2`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    Bigger Title
  </h3>
  <p>
    Better Content
  </p>
</div>
`;
